### PR TITLE
Feature: circle config

### LINKED ISSUE #11: Circle CI integration

### PR DESCRIPTION
**Pull request by jamesjwarren**
Fixes #11

### CODE REVIEW

File: circle.yml [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+  override:
+    - npm install
+
+# Test Commands
+
+test:
+  override:
+    - npm run lint
+    - npm run test
+    - npm run build
```
Comment (edwardoparearyee) [inline circle.yml]: don't think we need to run the build during the test commands, in fact `npm test` will run `npm pretest` which builds the app for the tests.

File: circle.yml [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+  post:
+    - ./node_modules/coveralls/bin/coveralls.js < coverage/lcov.info
+
+# Deployment
+
+deployment:
+  prod:
+    branch: master
+    commands:
+      - docker build -t $REPO:$TAG .
```
Comment (edwardoparearyee) [inline circle.yml]: in the deployment section you can run `npm run build`

File: circle.yml (hunk 1 above)
Comment (jamesjwarren) [inline circle.yml]: I had put the build here so we didn't have to run it in all deploy steps but yeh I agree doesn't make much sense I'll move it.

File: circle.yml [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+# CircleCI Configuration
+#
+
+# Machine Settings
+
+machine:
+  services:
+    - docker
+  node:
+    version: 5.6
```
Comment (edwardoparearyee) [inline circle.yml]: can we test on 4.3 a.k.a stable/LTS as well?

File: circle.yml [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+    - npm install
+
+# Test Commands
+
+test:
+  override:
+    - npm run lint
+    - npm run test
+  post:
+    - ./node_modules/coveralls/bin/coveralls.js < coverage/lcov.info
```
Comment (edwardoparearyee) [inline circle.yml]: not a big deal but if we install coveralls globally you can just run `coveralls < coverage/lcov.info`

**Comment by jamesjwarren** [comment]
@edwardoparearyee should be ready now, testing on both Node v4.4 and node v5.7

**Comment by edwardoparearyee** [comment]
:+1: 

It's a shame it's not as simple as travis to run tests on different versions of node

**Pull request merged**